test/unit/krane/kubernetes_resource/daemon_set_test.rb

```diff
@@ -107,6 +107,19 @@ def test_deploy_fails_when_not_all_pods_ready
     refute_predicate(ds, :deploy_succeeded?)
   end
 
+  def test_deploy_waits_for_daemonset_status_to_converge_to_pod_states
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+      "numberReady": 0,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    ready_pod_template = load_fixtures(filenames: ['daemon_set_pods.yml']).first # should be a pod in `Ready` state
+    node_templates = load_fixtures(filenames: ['nodes.yml'])
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: [ready_pod_template], node_templates: node_templates)
+    refute_predicate(ds, :deploy_succeeded?)
+  end
```

Comment:
Testing this against master yields a failure, proving my hypothesis that the divergent state between pods and the DS were the root of the error